bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java

```diff
@@ -28,55 +37,77 @@
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
-public class SitemapProviderImpl implements SitemapProvider {
+@Component(service = SitemapProvider.class)
+public class SitemapProviderImpl implements SitemapProvider, ModelRepositoryChangeListener {
 
-    protected static final String SITEMAP_FILEEXT = ".sitemap";
+    private static final String SITEMAP_MODEL_NAME = "sitemap";
+    protected static final String SITEMAP_FILEEXT = "." + SITEMAP_MODEL_NAME;
 
     private final Logger logger = LoggerFactory.getLogger(SitemapProviderImpl.class);
 
     private ModelRepository modelRepo = null;
 
+    private final Map<String, Sitemap> sitemapModelCache = new ConcurrentHashMap<>();
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC)

```

Comment:
Isn't this the default anyway and we agreed on not putting default values here? So I guess `@Reference` should be the same, but this is a minor issue.